### ISSUE TITLE
Update all documentation links to WCAG 2.2 equivalent

### ISSUE THREAD

**Issue by lb-** [jsx-eslint/eslint-plugin-jsx-a11y#1015]
It would be good for all the links to the WCAG website be updated to the equivalent WCAG 2.2 standard.

This came out just [over a year ago now](https://www.w3.org/TR/WCAG22/) and should be the canonical links we provide when giving guidance to developers.

There was this issue https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/1008 - however it was more about rules that need to change.

I am only suggesting we change our links in the documentation.

**Comment by ljharb** [comment]
The links in docs should match what we implement; we shouldn’t link to 2.2 unless we comply with it.

**Comment by lb-** [comment]
As per https://www.w3.org/WAI/standards-guidelines/wcag/new-in-22/#changes-from-wcag-21-to-wcag-22

> WCAG 2.2 provides 9 additional success criteria since WCAG 2.1. They are introduced on this page.
> 
> The 2.0 and 2.1 success criteria are essentially the same in 2.2, with one exception: 4.1.1 Parsing is obsolete and removed from WCAG 2.2. More information is in [WCAG 2 FAQ, 4.1.1 Parsing](https://www.w3.org/WAI/standards-guidelines/wcag/faq/#parsing411). WCAG 2.2 includes Notes about different languages; more information is in [WCAG 2 FAQ, internationalization](https://www.w3.org/WAI/standards-guidelines/wcag/faq/#i18n22).

I have noticed that some links have better linked examples and explanations but the actual success criteria itself remains the same.
